contracts/components/node_runners/NodeRunnerRegistryCore.sol

```diff
@@ -0,0 +1,379 @@
+    }
+
+
+    // ************* Scanner ownership *************
+
+    /**
+     * @notice Checks if scanner address has been registered
+     * @param scanner address.
+     * @return true if scanner is registered, false otherwise.
+     */
+    function isScannerRegistered(address scanner) public view returns (bool) {
+        return _scannerNodes[scanner].registered;
+    }
+
+    /**
+     * @notice Checks if scanner address has been registered to a specific nodeRunnerId
+     * @param scanner address.
+     * @param nodeRunnerId ERC721 token id of the Node Runner.
+     * @return true if scanner is registered to nodeRunnerId, false otherwise.
+     */
+    function isScannerRegisteredTo(address scanner, uint256 nodeRunnerId) public view returns (bool) {
+        return _scannerNodeOwnership[nodeRunnerId].contains(scanner);
+    }
+
+    /**
+     * @notice Method to register a Scanner Node and associate it with a nodeRunnerId. Before executing this method,
+     * register a scanner with Forta Scan Node CLI and obtain the parameters for this methods by executing forta auth.
+     * Follow the instructions here https://docs.forta.network/en/latest/scanner-quickstart/
+     * Individual ownership of a scaner node is not transferrable.
+     * A scanner node can be disabled, but not unregistered
+     * @param req ScannerNodeRegistration struct with the Scanner Node data.
+     * @param signature ERC712 signature, result from signed req by the scanner.
+     */
+    function registerScannerNode(ScannerNodeRegistration calldata req, bytes calldata signature) external onlyOwnerOf(req.nodeRunnerId) {
+        if (req.timestamp + registrationDelay < block.timestamp) revert RegisteringTooLate();
+        if (isScannerRegistered(req.scanner)) revert ScannerExists(req.scanner);
+        if (
+            !SignatureCheckerUpgradeable.isValidSignatureNow(
+                req.scanner,
+                _hashTypedDataV4(
+                    keccak256(abi.encode(_SCANNERNODEREGISTRATION_TYPEHASH, req.scanner, req.nodeRunnerId, req.chainId, keccak256(abi.encodePacked(req.metadata)), req.timestamp))
+                ),
+                signature
+            )
+        ) revert SignatureDoesNotMatch();
+        _scannerNodes[req.scanner] = ScannerNode(true, false, req.nodeRunnerId, req.chainId, req.metadata);
+        if (!_scannerNodeOwnership[req.nodeRunnerId].add(req.scanner)) revert ScannerAlreadyRegisteredTo(req.scanner, req.nodeRunnerId);
+        emit ScannerUpdated(scannerAddressToId(req.scanner), req.chainId, req.metadata, req.nodeRunnerId);
+    }
+
+    /**
+     * @notice Method to update a registered Scanner Node metadata string. Only the Node Runner that owns the scanner can update.
+     * @param nodeRunnerId ERC721 token id of the Node Runner.
+     * @param scanner address.
+     * @param metadata IPFS string pointing to Scanner Node metadata.
+     */
+    function updateScannerMetadata(
+        uint256 nodeRunnerId,
```

Comment:
Do we need this as an argument?

Reply:
Turns out no! I wrote this method before adding `nodeRunnerId` to `ScannerNode` struct. We don't need it now, plus we don't need to check if `ScannerNotRegisteredTo`, since the relationship is implicit and it would trigger `SenderNotOwner` now.

Thanks.

6871b19